src/Psalm/Internal/Analyzer/Statements/Expression/SimpleTypeInferer.php

```diff
@@ -109,6 +112,17 @@ public static function infer(
                 );
             }
 
+            if ($stmt instanceof PhpParser\Node\Expr\BinaryOp\ShiftLeft
+                || $stmt instanceof PhpParser\Node\Expr\BinaryOp\ShiftRight
+            ) {
+                return new Type\Union(
+                    [
+                        new Type\Atomic\TPositiveInt(),
```

Comment:
Shifts are perfectly capable to produce negative integers: https://3v4l.org/HXdvp

Reply:
Thanks!